common/src/registers/MachRegister.C

```diff
@@ -8,19 +8,24 @@
 
 #include <cassert>
 #include <unordered_map>
+#include <map>
+#include <vector>
 
 namespace {
   std::unordered_map<signed int, std::string> names;
   const std::string invalid_reg_name{"<INVALID_REG>"};
+  std::map<Dyninst::Architecture, std::vector<Dyninst::MachRegister>> all_regs;
 }
 
 namespace Dyninst {
 
   MachRegister::MachRegister() : reg(0) {}
 
-  MachRegister::MachRegister(signed int r) : reg(r) {}
+  MachRegister::MachRegister(signed int r) : reg(r) {
+    all_regs[getArchitecture()].push_back(*this);
```

Comment:
`MachRegister(Signed int r)` is used to construct a `MachRegister` from an id in more places than from a register definitions(`DEF_REGISTER`), for example `MachRegister::getBaseRegister`.  Moving the update to `all_regs` to `MachRegister(signed int r, std::string n)` would only add a new registers when a name is being added by `DEF_REGISTER`.

Reply:
Bah. Yes, of course.